### PR TITLE
[SPARK-28804][DOCS][SQL] Document DESCRIBE QUERY in SQL Reference

### PR DESCRIPTION
**Pull request by dilipbiswal**
### What changes were proposed in this pull request?
Document DESCRIBE QUERY statement in SQL Reference Guide. 

### Why are the changes needed?
Currently Spark lacks documentation on the supported SQL constructs causing
confusion among users who sometimes have to look at the code to understand the
usage. This is aimed at addressing this issue.

### Does this PR introduce any user-facing change?
Yes. 

**Before:**
There was no documentation for this.

**After.**
Screen Shot 2019-08-29 at 5 47 51 PM
Screen Shot 2019-08-29 at 5 48 06 PM
Screen Shot 2019-08-29 at 5 48 18 PM

### How was this patch tested?
Tested using jykyll build --serve

### CODE REVIEW

**Comment by dilipbiswal** [comment]
cc @gatorsmile

File: docs/sql-ref-syntax-aux-describe-query.md [hunk 1]
```diff
@@ -19,4 +19,34 @@ license: |
   limitations under the License.
 ---
 
-**This page is under construction**
+### Description
+The DESCRIBE QUERY statement is used to return the metadata of output
```
Comment (srowen) [inline docs/sql-ref-syntax-aux-describe-query.md]: Back tick the reserved words

File: docs/sql-ref-syntax-aux-describe-query.md [hunk 2]
```diff
@@ -19,4 +19,34 @@ license: |
...
+* For detailed syntax of query parameter please refer to [select-statement](sql-ref-syntax-qry-select.html)
+  and the examples below.
+* DESC and DESCRIBE are interchangeble and mean the same thing.
+
+### Examples
+{% highlight sql %}
+-- Returns column name, column type and comment info for a simple select query
+DESCRIBE QUERY select * customer;
+-- Returns column name, column type and comment info for a inline table.
+DESC QUERY VALUES(1.00D, 'hello') as test_tab(c1, c2);
```
Comment (srowen) [inline docs/sql-ref-syntax-aux-describe-query.md]: I don't know SQL well enough to know: you can write 1.00D as a literal? the "D" works?

**Review by srowen — CHANGES_REQUESTED** [review]

File: docs/sql-ref-syntax-aux-describe-query.md (hunk 2 above)
Comment (dilipbiswal) [reply docs/sql-ref-syntax-aux-describe-query.md]: @srowen Yeah :-). Here is the grammar snippet for a double literal.
```
DOUBLE_LITERAL
    : DIGIT+ EXPONENT? 'D'
    | DECIMAL_DIGITS EXPONENT? 'D' {isValidDecimal()}?
```

**Comment by srowen** [comment]
PS you might look at the other similar PRs to add these docs to try to harmonize the style and format a little bit. They're pretty similar. That can mean reviewing others' changes.

**Comment by gatorsmile** [comment]
LGTM

Thanks! Merged to master.

**Comment by dilipbiswal** [comment]
@gatorsmile Thank you for fixing the mistake and merging.

**Pull request closed without merging**